### PR TITLE
Rel

### PR DESCRIPTION
**Pull request by clayrat**

### CODE REVIEW

**Comment by clayrat** [comment]
Ready for review.

File: src/Rel.lidr [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+
+> le_not_a_partial_function : Not (Partial_function Le)
+> le_not_a_partial_function f = absurd $ f 0 0 1 Le_n (Le_S Le_n)
+
+
+==== Exercise: 2 stars, optional
+
+\ \todo[inline]{Again, "earlier" = \idr{IndProp}}
+
+Show that the idr{Total_relation} defined in earlier is not a partial function.
```
Comment (yurrriq) [inline src/Rel.lidr]:
```tex
\idr{Total_relation}
```

File: src/Rel.lidr [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+
+> -- FILL IN HERE
+
+$\square$
+
+
+==== Exercise: 2 stars, optional
+
+Show that the idr{Empty_relation} that we defined earlier is a partial function.
```
Comment (yurrriq) [inline src/Rel.lidr]:
```tex
\idr{Empty_relation}
```

File: src/Rel.lidr [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+coincides with the \idr{Le} relation.
+
+\todo[inline]{Copied `<->` for now}
+
+> iff : {p,q : Type} -> Type
+> iff {p} {q} = (p -> q, q -> p)
+>
+> syntax [p] "<->" [q] = iff {p} {q}
+
+> next_nat_closure_is_le : (n <=' m) <-> (Clos_refl_trans Next_nat n m)
```
Comment (yurrriq) [inline src/Rel.lidr:314]: Note for later: minted handles this definition poorly. See also #30.

**Review by yurrriq — CHANGES_REQUESTED** [review]
We'll need to add `Rel` to `software_foundations.ipkg` and `Rel.lidr` to `src/Makefile` too.

**Comment by yurrriq** [comment]
Other than those little comments this looks good to me.

**Comment by yurrriq** [comment]
Feel free to merge this one yourself after those trivial changes.

File: src/Rel.lidr (hunk 3 above)
Comment (clayrat) [reply src/Rel.lidr:314]: So what should we do here? Add a `todo`?

**Comment by yurrriq** [comment]
Sure, though linking the relevant issue in my previous  comment might be enough. This is a toolchain issue rather than a code one.

**Comment by clayrat** [comment]
Let's merge `ProofObjects` first maybe and add to `.ipkg`/`Makefile` after that? I feel like it'll be a merge conflict anyway.

**Review by yurrriq — APPROVED** [review]

**Pull request merged**